pandas/core/indexes/base.py

```diff
@@ -4947,9 +4949,9 @@ def slice_indexer(self, start=None, end=None, step=None, kind=None):
         start_slice, end_slice = self.slice_locs(start, end, step=step, kind=kind)
 
         # return a slice
-        if not is_scalar(start_slice):
+        if not is_scalar(Fraction(start_slice)):
```

Comment:
Can you explain why we're trying to turn this into a fraction? I think this may only be working by accident since TypeErrors are getting caught upstream.